infrastructure/src/main/java/org/corfudb/infrastructure/ManagementServer.java

```diff
@@ -339,6 +343,52 @@ public synchronized void handleFailureDetectedMsg(CorfuPayloadMsg<FailureDetecto
     }
 
     /**
+     * Adds a new node to the layout.
+     *
+     * @param msg corfu message containing ADD_NODE_REQUEST
+     * @param ctx netty ChannelHandlerContext
+     * @param r   server router
+     */
+    @ServerHandler(type = CorfuMsgType.ADD_NODE_REQUEST, opTimer = metricsPrefix + "add-node")
+    public void handleAddNodeRequest(CorfuPayloadMsg<AddNodeRequest> msg,
+                                     ChannelHandlerContext ctx, IServerRouter r,
+                                     boolean isMetricsEnabled) {
+        if (isShutdown()) {
```

Comment:
Shouldn't this check validate this conditions on all servers?

Also, it seems redundant to check  if this condition is true from the handler. AbstractServer seems a more appropriate place to make this check.

Reply:
Yes, this and the bootstrapping check should be done at a higher layer. Will move it.